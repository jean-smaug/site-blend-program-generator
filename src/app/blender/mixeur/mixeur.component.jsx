// @flow

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { ToastContainer, ToastMessage } from 'react-toastr';
import { filterConferences, orderConferences, filterByTags } from '../../../lib/dataFilter.lib';
import { getConferences, writeStore } from '../../../lib/database';
import { mixConferencesAction } from '../../smoothie/smoothie.action';
import { Conferences } from '../../smoothie/smoothie.type';
import { setConferencesStore, setKeyStore, remove, isStore } from '../../../lib/localStorage.lib';

/**
 * Component for Submit button to mix
 */
export class MixeurComponent extends Component {
  props: {
    addConference: (orderedConferences: Conferences) => void,
    form: Object,
  };

  submitBtn = async () => {
    if (this.props.form.informations.isValidEmail) {
      const conferences = (await getConferences()) || [];
      const { addConference, form } = this.props;
      const orderedConferences = orderConferences(
        filterConferences(conferences, form.domains, form.keywords),
      );
      console.log('tags', filterByTags(conferences, form.keywords));
      console.log('tags', conferences, form.keywords);

      // setKeyStore(await writeStore({ smoothie: orderedConferences, blender: form }));
      // if (isStore('isAlreadyShow')) remove('isAlreadyShow');
      // addConference(orderedConferences);
      // setConferencesStore({
      //   dayOne: orderedConferences.dayOne,
      //   dayTwo: orderedConferences.dayTwo,
      // });
    } else {
      this.toastError.error(
        "L'email que vous avez renseign√© a un format incorrect.",
        'Attention !',
        {
          timeOut: 7000,
          extendedTimeOut: 1000,
          closeButton: true,
        },
      );
    }
  };

  render() {
    return (
      <div>
        <ToastContainer
          ref={(input) => {
            this.toastError = input;
          }}
          toastMessageFactory={React.createFactory(ToastMessage.animation)}
          className="toast-top-full-width"
        />
        <input className="mix-btn" type="button" onClick={this.submitBtn} value="GO MIX" />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  form: state.form,
});

const mapDispatchToProps = dispatch => ({
  addConference: (word) => {
    dispatch(mixConferencesAction(word));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(MixeurComponent);
