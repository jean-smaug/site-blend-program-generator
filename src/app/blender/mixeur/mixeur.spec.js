import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { MixeurComponent } from './mixeur.component';

describe('blender.component', () => {
  let wrapper;
  let store;
  const mockStore = configureStore();
  const initialState = {};

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<MixeurComponent store={store} />);
  });

  it('should render blender component', () => {
    expect(wrapper.getNodes()).toMatchSnapshot();
  });
});
