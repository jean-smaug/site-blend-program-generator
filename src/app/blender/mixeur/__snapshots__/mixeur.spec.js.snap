// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blender.component should render blender component 1`] = `
Array [
  <div>
    <ToastContainer
        className="toast-top-full-width"
        id="toast-container"
        newestOnTop={true}
        onClick={[Function]}
        preventDuplicates={true}
        toastMessageFactory={[Function]}
        toastType={
            Object {
                "error": "error",
                "info": "info",
                "success": "success",
                "warning": "warning",
              }
        }
    />
    <input
        className="btn-begin"
        onClick={[Function]}
        type="button"
        value="Voir mon planning >"
    />
</div>,
]
`;
